File: tests/fixtures/generic/vulnerable/taint_deserialization.js
// @audit-fixture
// @rule: taint_deserialization
// @expected: detected
const serialize = require("node-serialize");
const express = require("express");
const app = express();
app.post("/load", (req, res) => {
    const data = req.body.data;
    const obj = serialize.unserialize(data);
    res.json(obj);
});

File: tests/fixtures/generic/vulnerable/taint_deserialization.cs
// @audit-fixture
// @rule: taint_deserialization
// @expected: detected
using System.Runtime.Serialization.Formatters.Binary;
public class DataController : Controller {
    public IActionResult Load() {
        var formatter = new BinaryFormatter();
        var obj = formatter.Deserialize(Request.Body);
        return Ok(obj);
    }
}
